pages/components/Todolist.tsx

```diff
@@ -14,13 +14,22 @@ const Todolist = () => {
     const [isModalOpen, setIsModalOpen] = useState(false);
     const [todoToDelete, setTodoToDelete] = useState<number | null>(null);
 
+    useEffect(() => {
+        const storedTodos = localStorage.getItem('todos');
+        if (storedTodos) {
+            setTodos(JSON.parse(storedTodos));
+        }
+    }, []);
+
     const addTodo = () => {
         if (newTodo.trim()) {
             const newTask = {
                 text: newTodo,
                 status: 'pending',
             };
-            setTodos([...todos, newTask]);
+            const updatedTodos = [...todos, newTask];
+            setTodos(updatedTodos);
+            localStorage.setItem('todos', JSON.stringify(updatedTodos));  // Save to localStorage
```

Comment:
when you find yourself repeating something over 2-3 times within the same code base - ask yourself "can I make this reusable? Either as a variable or function"